includes/class-wc-stripe-api.php

```diff
@@ -123,14 +123,31 @@ public static function get_headers() {
 		$user_agent = self::get_user_agent();
 		$app_info   = $user_agent['application'];
 
-		$headers = apply_filters(
+		$headers = apply_filters_deprecated(
 			'woocommerce_stripe_request_headers',
 			[
-				'Authorization'  => 'Basic ' . base64_encode( self::get_secret_key() . ':' ),
-				'Stripe-Version' => self::STRIPE_API_VERSION,
-			]
+				[
+					'Authorization'  => 'Basic ' . base64_encode( self::get_secret_key() . ':' ),
+					'Stripe-Version' => self::STRIPE_API_VERSION,
+				],
+			],
+			'9.6.0',
+			'wc_stripe_request_headers',
+			'The woocommerce_stripe_request_headers filter is deprecated since WooCommerce Stripe Gateway 9.6.0, and will be removed in a future version. Use wc_stripe_request_headers instead.'
 		);
 
+		/**
+		 * Filters the request headers sent to the Stripe API.
+		 *
+		 * @since 9.6.0
+		 *
+		 * @param array $headers The headers from the deprecated filter.
```

Comment:
We should only document that these are the default headers we send.
```suggestion
		 * @param array $headers The default headers we send to the Stripe API.
```